### ISSUE TITLE
Training BERT with Soft Labels

### ISSUE THREAD

**Issue by Mahhos** [google-research/bert#1165]
Hi. I need to train BERT using soft labels (e.g. class probabilities [0.4, 0.6]) instead of using hard labels (0 or 1). I need to do this in iterations where I predict logits for an unlabeled set and add these pseudo labels to the training set (which has soft labels) and again train another model. The second part is easy. However, I am not sure how to train BERT with soft labels on first hand.

**Comment by bhavishya-pohani** [comment]
Even your normal labels are converted to one hot vectors, instead of those one hot vectors use the probabilities.

**Comment by Mahhos** [comment]
You mean I can normally input probabilities instead of hard labels in my train.tsv and dev.tsv files and the program should be able to handle them?

**Comment by bhavishya-pohani** [comment]
No, that would not work, you need to change your code base to handle that, places like `create_examples`/anywhere else where a hard label is expected, change it to expect a vector instead.
